File: contest_05/01/main.Getacat.cpp
#include <iostream>

class Cat {
    public:
        Cat(bool status) {
            alive = status;
        }

        bool is_alive() {
            return alive;
        }
    private:
        bool alive;
};

class Box {
    public:
        Cat open() {
            return Cat(rand() > (RAND_MAX / 2));
        }
};

int main() {
    Box box;
    Cat cat = box.open();
    if (cat.is_alive()) {
        std::cout << "alive";
    } else {
        std::cout << "dead";
    }
    return 0;
}

File: contest_05/06/main.Getacat.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

type Box struct {
}

type Cat struct {
	alive bool
}

func (b Box) open() Cat {
	rand.Seed(time.Now().UnixNano())
	return Cat{rand.Intn(2) == 1}
}

func (c Cat) is_alive() bool {
	return c.alive
}

func main() {
	var box Box
	cat := box.open()
	if cat.is_alive() {
		fmt.Print("alive")
	} else {
		fmt.Print("dead")
	}
}
